mypy/semanal_pass1.py

```diff
@@ -74,6 +74,10 @@ def visit_file(self, file: MypyFile, fnam: str, mod_id: str, options: Options) -
                     if last.end_line is not None:
                         # We are on a Python version recent enough to support end lines.
                         self.skipped_lines |= set(range(next_def.line, last.end_line + 1))
+                    # To avoid running a new visitor here, let's assume nobody does
+                    # `assert sys.platform == 'foo'; import unknown` on one line
+                    # at top level.
+                    file.imports = [i for i in file.imports if i.line < next_def.line]
```

Comment:
I think you can maybe check `(line, col)` to avoid the comment?

Reply:
Yep, thanks!